epix/io.py

```diff
@@ -80,75 +81,106 @@ def loader(directory,
         arg_debug (bool): If true, print out loading information.
         outer_cylinder (dict): If specified will cut all events outside of the
             given cylinder.
-        kwargs_uproot_arrays (dict): Keyword arguments passed to .arrays of
-            uproot4.
+        kwargs (dict): Keyword arguments passed to .arrays of uproot4.
         cut_by_eventid (bool): If true event start/stop are applied to
             eventids, instead of rows.
 
     Returns:
         awkward1.records: Interactions (eventids, parameters, types).
         integer: Number of events simulated.
     """
-    ttree, n_simulated_events = _get_ttree(directory, file_name)
+    file = os.path.join(directory, file_name)
 
-    if arg_debug:
-        print(f'Total entries in input file = {ttree.num_entries}')
-        cutby_string='output file entry'
-        if cut_by_eventid:
-            cutby_string='g4 eventid'
-
-        if kwargs_uproot_arrays['entry_start'] is not None:
-            print(f'Starting to read from {cutby_string} {kwargs_uproot_arrays["entry_start"]}')
-        if kwargs_uproot_arrays['entry_stop'] is not None:
-            print(f'Ending read in at {cutby_string} {kwargs_uproot_arrays["entry_stop"]}')
-
-    # If user specified entry start/stop we have to update number of
-    # events for source rate computation:
-    if kwargs_uproot_arrays['entry_start'] is not None:
-        start = kwargs_uproot_arrays['entry_start']
-    else:
-        start = 0
-
-    if kwargs_uproot_arrays['entry_stop'] is not None:
-        stop = kwargs_uproot_arrays['entry_stop']
-    else:
-        stop = n_simulated_events
-    n_simulated_events = stop - start
-
-    if cut_by_eventid:
-        # Start/stop refers to eventid so drop start drop from kwargs
-        # dict if specified, otherwise we cut again on rows.
-        kwargs_uproot_arrays.pop('entry_start', None)
-        kwargs_uproot_arrays.pop('entry_stop', None)
-
-    # Columns to be read from the root_file:
-    column_names = ["x", "y", "z", "t", "ed",
-                    "type", "trackid",
-                    "parenttype", "parentid",
-                    "creaproc", "edproc"]
-
-    # Conversions and parameters to be computed:
-    alias = {'x': 'xp/10',  # converting "geant4" mm to "straxen" cm
-             'y': 'yp/10',
-             'z': 'zp/10',
-             'r': 'sqrt(x**2 + y**2)',
-             't': 'time*10**9'
-             }
+    if not os.path.isfile(file):
+        print("File does not exist!")
+        #cancel epix execution here
```

Comment:
I think in that case you should raise an error is not it? Although I am not sure if this is really needed. I would suspect that the inhouse methods of uproot and pandas have this already. So maybe it is better to remove this if-statement?